### ISSUE TITLE
Category based scoring (Single-handed, Women, Masters, &c.)

### ISSUE THREAD

**Issue by jbfraser1** [SailScores/SailScores#150]
When scoring some series and many (most?) regattas, sailors may belong to a special division. For example, "Women" or "Single-handed" sailors may be tracked for scoring in addition to the being part of the usual fleet scoring.

SailScores should allow sailors to be added to a division and for series/regatta scoring to indicate/highlight the standings of those competitors.

Thought needs to be given to the case where a sailor may belong to a division for one event but not others. (Probably a property of the series/regatta.)

**Comment by jbfraser1** [comment]
Continuing concept development on this. Does this need to support categories where the competitor may be in that category for only some races of the series: for example: "Single handed" should count all of a competitors results when they sail single handed, but exclude results for races they sail with crew.

For now, I don't think I'll build this option in, but I want to give thought to how this might happen in the future.

**Comment by jbfraser1** [comment]
Planning on starting work on this with "Filtered Series." This would be an advanced/hidden option for a series. When selected, the creator/scorekeeper would need to choose the included Categories for that series. Then, to have scores showing in that series, a race would need to be assigned to the series and a competitor would need to match one of the categories.

This would work with the current regatta set up for many categories. A regatta could include a typical series with all competitors and also include a "Youth Series" which would filter the race results down to just competitors tagged as Youth.

This could support competitors that age out of youth pretty easily, but would not easily support criteria that changed from race to race, like the previously mentioned "Singled Handed" scenario.

A work-around for that for now would be creating a Single Handed series and created races that only included single handed sailors. A little more manual work, but seems pretty intuitive to me.

**Comment by OrthoLoess** [comment]
It would be useful to be able to apply a division to a particular race/races in a series, e.g. during a regatta, the morning dinghy series includes a special trophy for junior competitors in the race on Thursday.

I would not consider this a priority and the club I sail at that actually has this level of complexity in its calendar currently uses Halsail. The club I desperately need a better system for does not need this level of detail at present.

**Comment by jbfraser1** [comment]
> [...] I would not consider this a priority and the club I sail at that actually has this level of complexity in its calendar currently uses Halsail. [...]

Makes sense. It seems like the categories are more important for regattas, and not frequently used in club scoring, which is core to SailScores.

**Comment by jbfraser1** [comment]
Starting work on adding a Fleet to series.

This would be an optional setting, with empty being shown as "-Any-". If not set, functionality would be like today: any race assigned to the series would count as results. If not set to "-Any-"(empty value) then an existing fleet would need to be selected (dropdown.) Dropdown should only show active fleets, but inactive fleets would need to be saved on series edit.

If a fleet is selected for a series, another option should be available:
- Use "All Competitor Race Score" when scoring

This defaults to false. If true, the scoring of the series will calculate using the finishing place in races including all results. (If the first person in the Single-Handed fleet finished in third place overall in the race, true on this setting would use 3, false on this setting would use 1.

**Comment by jbfraser1** [comment]
Coming soon...

<img width="660" height="304" alt="Image" src="https://github.com/user-attachments/assets/84aba665-bf0d-4c65-b655-d741e9816cf2" />

**Comment by jbfraser1** [comment]
The variety of fleets (All boats in club, selected competitors, and specific boat classes.) are creating a lot of corner cases for this. I'm working through them, and really expect to get this deployed to prod in the next few days.

**Comment by jbfraser1** [comment]
Deployed to prod.
